notification/store.go

```diff
@@ -29,6 +29,9 @@ type Store interface {
 	VerifyContactMethod(ctx context.Context, cmID string, code int) error
 	Code(ctx context.Context, id string) (int, error)
 	FindManyMessageStatuses(ctx context.Context, ids ...string) ([]MessageStatus, error)
+
+	// LastMessageStatus will return the MessageStatus and creation timestamp of the message matching the filter critera.
+	LastMessageStatus(ctx context.Context, typ string, cmID string, from time.Time) (*MessageStatus, time.Time, error)
```

Comment:
Use a discrete type instead of arbitrary string for the `typ` parameter.

Reply:
I couldn't use the discrete type, MessageType from notification, because of the same reason above